### PR TITLE
WIP Citizenship presence/visa held requirement calculation

### PR DESCRIPTION
**Pull request by dlouise64**
Added presence rules for NZ Citizenship.

Currently waiting for DAY support

requires https://github.com/openfisca/openfisca-core/pull/766

### CODE REVIEW

**Comment by Br3nda** [comment]
@dlouise64 you might need to check your editor is doing space and not tabs

**Comment by dlouise64** [comment]
@Br3nda yep we can wait for Days 👍

File: openfisca_aotearoa/variables/acts/citizenship/citizenship.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+from openfisca_core.model_api import *
+from openfisca_aotearoa.entities import Person
+
+
+class citizenship__present_for_min_1350_days_during_5_years(Variable):
+    value_type = bool
+    entity = Person
+    definition_period = MONTH
+    label = u"Applicant was present in New Zealand for a min of 1,350 days during the 5 years immediately preceding the date of application"
+    reference = "http://www.legislation.govt.nz/act/public/1977/0061/latest/DLM443855.html"
```
Comment (Br3nda) [inline openfisca_aotearoa/variables/acts/citizenship/citizenship.py]: We'll need to add the calculation here -- openfisca has the days they were here and the days they weren't, and needs to work out if that was enough

File: openfisca_aotearoa/tests/citizenship/citizenship.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        "Not eligible 240 days",
+        "Eligible"
+      ]
+
+  output_variables:
+    citizenship__eligibility:
+      2018-08:
+        - false # Not eligible 1350 days
+        - false # Not eligible 240 days
+        - true  # Eligible
```
Comment (Br3nda) [inline openfisca_aotearoa/tests/citizenship/citizenship.yaml]: some tests like this

```
... _present_in_nz:
  2018-01-01: true
  2018-01-02: true
  2018-01-03: true
  2018-01-04: true
  2018-01-05: true
  2018-01-06: true
  2018-01-07: true
  2018-01-08: false
  2018-01-08: false
  2018-01-10: true
```

**Comment by Br3nda** [comment]
instructions for running (until we get the PR merged in OF)

```
cd openfisca-aotearoa
pew rm openfisca
pew new openfisca --python=python3.7
pip install "."
pip install -e "git+git://github.com/ServiceInnovationLab/openfisca-core.git@period-day#egg=openfisca_core"
```
then you can run the tests

```
openfisca-run-test --country openfisca_aotearoa openfisca_aotearoa/tests
```

File: .circleci/config.yml [hunk 3]
```diff
@@ -92,6 +93,7 @@ jobs:
             pip install --upgrade pip twine wheel
             # pip install --editable git+https://github.com/openfisca/openfisca-core.git@BRANCH#egg=OpenFisca-Core[web-api]  # use a specific branch of OpenFisca-Core
             pip install --editable .[dev] --upgrade
+            pip install -e "git+git://github.com/ServiceInnovationLab/openfisca-core.git@period-day#egg=openfisca_core"
```
Comment (Br3nda) [inline .circleci/config.yml]: same here. needs removal before we merge. otherwise it'll break on prod

**Pull request closed without merging**